std/json.d

```diff
@@ -1252,7 +1252,8 @@ string toJSON(const ref JSONValue root, in bool pretty = false, in JSONOptions o
                     }
                 }
                 else {
-                    json.put(to!string(val));
+                    import std.format : format;
+                    json.put("%.9g".format(val));
```

Comment:
This will cause a massive *loss* of precision, because `val` here is always a 64-bit `double`. (See the definition of `JSONValue` starting on line 100.)

I notice also that most of the unit tests below are improperly written to test with `float` values instead of `double`.